engine/lib/analyses/mutable_variables.ml

```diff
@@ -0,0 +1,91 @@
+open Base
+open Utils
+
+module%inlined_contents Make (F : Features.T) = struct
+  module FA = F
+  module A = Ast.Make (F)
+  open Ast
+
+  type pre_data =
+    ( concrete_ident,
+      concrete_ident list,
+      Concrete_ident.comparator_witness )
+    Map.t
+    * (concrete_ident, int, Concrete_ident.comparator_witness) Map.t
+
+  type analysis_data =
+    ( concrete_ident,
+      local_ident list * ((local_ident * A.ty) * int) list,
+      Concrete_ident.comparator_witness )
+    Map.t
```

Comment:
Same here.
Actually, maybe you could define a `'t analysis_data` of type `(local_ident list * 't) Map.M(Concrete_ident).t`?
In my opinion that would make the aim of the code more clear: your `analyse` function would just add information from a `unit analysis_data` to a `(TypedLocalIdent.t * int) list analysis_data`, wdyt?

(note `TypedLocalIdent` is defined in `Ast_utils`)

Reply:
It will not always be the case that the pre and post analysis data types can be described using the same dependent type. So I think it might be better to keep it general and have a type for both from and to.

Reply:
I added a comment about the `ANALYSIS` type module below: I'm not sure we should constraint the type signature of such analysis functions. (then `ANALYSIS` is basically encoding a function of arity two where the second argument is more or less fixed, it's not bringing much value)